src/modules/auth/components/ChooseAvatar/chooseAvatar.js

```diff
@@ -1,103 +1,113 @@
-import React from 'react';
-import { withTranslation } from 'react-i18next';
+import React, { useState, useEffect, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
 import { Link } from 'react-router-dom';
 import routes from 'src/routes/routes';
-import { PrimaryButton, TertiaryButton } from 'src/theme/buttons';
+import { PrimaryButton, TertiaryButton } from '@theme/buttons';
+import useSettings from '@settings/hooks/useSettings';
+import { extractKeyPair, extractAddressFromPublicKey } from '@wallet/utils/account';
+import { defaultDerivationPath } from '@account/const';
 import WalletVisual from '@wallet/components/walletVisual';
 import registerStyles from '../Signup/register.css';
 import styles from './chooseAvatar.css';
 
-class ChooseAvatar extends React.Component {
-  constructor() {
-    super();
+// eslint-disable-next-line max-statements
+const ChooseAvatar = ({ accounts, selected, handleSelectAvatar, nextStep }) => {
+  const [deselect, setDeselect] = useState({});
+  const [accountList, setAccountList] = useState([]);
+  const prevDeselect = useRef({});
+  const wrapperRef = useRef();
+  const timeout = useRef();
+  const { t } = useTranslation();
+  const { enableAccessToLegacyAccounts } = useSettings('enableAccessToLegacyAccounts');
```

Comment:
Since all new accounts are created with default custom derivation path, you do not need this logic, `enableAccessToLegacyAccounts` is false by default.

Reply:
Noticed a strange scenario which might affect the intended behaviour regarding this.
1. Go to http://0.0.0.0:8080/#/account/add/secret-recovery
2. Check `Enable access to legacy Lisk accounts` 
3. Go to http://0.0.0.0:8080/#/register?strength=128
4. Continue the flow and the user will encrypt the new account with this API `cryptography.legacy.getKeys(passphrase)`.

Reply:
@ManuGowda The `extractKeyPair` function requires `enableAccessToLegacyAccounts` and even if it's false by default, changing it affects the account generation. The bug occurred because it wasn't taken into account previously

Reply:
@oskarleonard I followed the steps and it's fine because prior to the SDK update, accounts were generated based on that function

Reply:
What i mean was that the account would in that case not be created with the default custom derivation Manu talked about, but instead with our legacy one. Which is why it wont work to remove it in this step.

If feels strange that we removed the `enableAccessToLegacyAccounts` from settings if it can be used in this way. Feels like ✅  "Enable access to legacy Lisk accounts" should not be preserved when creating a new account. Ie that the `enableAccessToLegacyAccounts` state should only be used when adding an account from Secret recovery phrase 🤔

Reply:
@oskarleonard new account created should always be using custom derivation path, ikem now has fixed it

Reply:
@ManuGowda It will still be possible to create a new account using this legacy API `cryptography.legacy.getKeys(passphrase)`. Is that intended?